slave/keithley/k2182.py

```diff
@@ -69,16 +69,106 @@ class Sense(Driver):
     :ivar function: The sense function, either 'temperature' or 'voltage' (dc).
+    :ivar nplc: Set integration rate in line cycles (0.01 to 60).
+    :ivar auto_range: Enable/Disable auto ranging
+    :ivar range: Set the measurement range (0 to 120 Volt)
 
     """
     def __init__(self, connection):
         super(Sense, self).__init__(connection)
+        self.delta_mode = Command(
+            ':SENS:VOLT:DELT?',
+            ':SENS:VOLT:DELT',
+            Boolean
+        )
         self.function = Command(
             ':SENS:FUNC?',
             ':SENS:FUNC',
             Mapping({'voltage': '"VOLT:DC"', 'temperature': '"TEMP"'})
         )
+        self.nplc = Command(
+            ':SENS:VOLT:NPLC?',
+            ':SENS:VOLT:NPLC',
+            Float(min=0.01, max=50)
+        )
+        self.auto_range = Command(
+            ':SENS:VOLT:RANG:AUTO?',
+            ':SENS:VOLT:RANG',
+            Boolean
+        )
+        self.range = Command(
+            ':SENS:VOLT:RANG?',
+            ':SENS:VOLT:RANG',
+            Float(min=0, max=120)
+        )
+
+
+class System(Driver):
+    """The System command layer.
+
+    :param connection: A connection object.
+
+    :ivar autozero: Enable/Disable autozero.
+    :ivar front_autozero: Enable/Disable front autozero (disable to speed
+                          up delta measurements)
+    """
+    def __init__(self, connection):
+        super(System, self).__init__(connection)
+        self.autozero = Command(
+            ':SYST:AZER?',
+            ':SYST:AZER',
+            Boolean
+        )
+        self.front_autozero = Command(
+            ':SYST:FAZ?',
+            ':SYST:FAZ',
+            Boolean
+        )
+
+    def preset(self):
+        """Return to system preset defaults."""
+        self._write(':SYST:PRES')
+
+
+
+class Trace(Driver):
+    """The Trace command layer.
+
+    :param connection: A connection object.
+
+    :ivar points: Specify number of readings to store (2-1024).
+    :ivar feed:   Source of readings ('sense', 'calculate' or `None`).
+    :ivar feed_control: Buffer control mode ('never' or 'next')
+    """
+    def __init__(self, connection):
+        super(Trace, self).__init__(connection)
+        self.points = Command(
+            ':TRAC:POIN?',
+            ':TRAC:POIN',
+            Integer(min=2, max=1024)
+        )
+        self.feed = Command(
+            ':TRAC:FEED?',
+            ':TRAC:FEED',
+            Mapping({'sense': 'SENS', 'calculate': 'CALC', None: 'NONE'})
+        )
+        self.feed_control = Command(
+            ':TRAC:FEED:CONT?',
+            ':TRAC:FEED:CONT?',
+            Mapping({'next': 'NEXT', 'never': 'NEV'})
+        )
+
+    def clear(self):
+        """Clear readings from buffer."""
+        self._write(':TRAC:CLEAR')
+
+    def free(self):
+        """Query bytes available and bytes in use."""
+        response = self._query((':TRAC:FREE?', String))
```

Comment:
This should work as well:

    return response = self._query((':TRAC:FREE?', [Float, Float]))